arrows/klv/klv_read_write.txx

```diff
@@ -55,6 +55,29 @@ KWIVER_ALGO_KLV_EXPORT
 bool
 _left_shift_overflow( T value );
 
+// ---------------------------------------------------------------------------
+template < class Float > union _float_int_union;
+
+// ---------------------------------------------------------------------------
+template <>
+union _float_int_union< float >
+{
+  float float_value;
+  uint32_t int_value;
```

Comment:
So... I know this is a "typical" way to do [type-punning](https://en.wikipedia.org/wiki/Type_punning), but as I understand it, absent [`std::bit_cast`](https://en.cppreference.com/w/cpp/numeric/bit_cast) (C++20), the only non-UB way to do this is with `memcpy` ([note](https://stackoverflow.com/a/31080901/1644352)). (Which is somewhat less elegant in some ways, but means you probably don't need this `union` mess, and the compiler should also optimize away the `memcpy`. In fact, "should" here is used in the sense "most folks, *including compiler vendors*, would consider it a compiler bug if it doesn't".)

(Just to keep things exciting, this is true *in C++*. A `union` *is* actually the correct method in C99 and later.)